draft-ietf-httpbis-digest-headers.md

```diff
@@ -358,152 +438,213 @@ Examples:
 ~~~ http-message
 Want-Digest: sha-256
 Want-Digest: sha-512;q=0.3, sha-256;q=1, unixsum;q=0
-Want-Content-Digest: sha-256
-Want-Content-Digest: sha-512;q=0.3, sha-256;q=1, unixsum;q=0
 ~~~
 
+## Want-Representation-Digest and Want-Content-Digest fields {#want-digest-sf}
 
-# Digest Algorithm Values {#algorithms}
+The `Want-Representation-Digest` field indicates the sender's desire to receive a
+representation digest on messages associated with the request URI and
+representation metadata, using the `Representation-Digest` field.
+
+The `Want-Content-Digest` field indicates the sender's desire to receive a content digest
+on messages associated with the request URI and representation metadata, using
+the `Content-Digest` field.
 
-Digest-algorithm values are used to indicate a specific digest computation.
+`Want-Representation-Digest` and `Want-Content-Digest` are Structured Fields
+List (see {{Section 3.1 of STRUCTURED-FIELDS}}).
 
 ~~~ abnf
-   digest-algorithm = token
+   Want-Representation-Digest = sf-list
+   Want-Content-Digest = sf-list
 ~~~
 
-All digest-algorithm token values are case-insensitive
-but lower case is preferred;
-digest-algorithm token values MUST be compared in a case-insensitive fashion.
+List items convey hashing algorithm preferences (see {{algorithms}}). Items can
+be parameterized with the `q` parameter, which can be used to express an
+ascending relative weighted preference between 0 and 1. `q` MUST be sf-decimal.
```

Comment:
Referencing the qvalue rule mentioned in Notational conventions.

```suggestion
List items convey hashing algorithm preferences (see {{algorithms}}).
Items can be parameterized with the `q` parameter, which can be used to express an
ascending relative weighted preference between 0 and 1.
`q` MUST be sf-decimal compatible to the "qvalue" rule.
```

Reply:
I think this is an area for some WG feedback. 

My approach here is to define a new parameter `q`. It's based on qvalue but it's compatibility is arguable, because the definition of compatibility is not well defined. If there are additional constraints for the parameter, we should state these explicitly. But we should also think about whether the expectations for acting on this `q` are as clear as qvalue.

If we assume that future SF-based fieIds might want a qvalue-like parameter. Perhaps we should consider making this document the canonical definition of `q` parameter. (it could be a standalone document too but there's more bootstrapping to that).

Reply:
> this document the canonical definition of q parameter

@reschke WDYT about SF and qvalue?